[KS2499] iP

KS2499 opened the pull request:
# DukePro

> "Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

### DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

### All you need to do is,

1. download it from [here](https://github.com/KS2499/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

### And it is **FREE**!

### Features:
- [x] Managing Tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:


```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

KS2499 committed:
Greet, Echo, Exit

KS2499 committed:
Level 2. Add, List

KS2499 committed:
Level 3. Mark as Done

KS2499 committed:
Level 4. ToDos, Events, Deadlines

KS2499 committed:
Automated Text UI Testing

KS2499 committed:
Level 5. Handle Errors

KS2499 committed:
Level 6. Delete

Beebeeoii commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+            case BYE:
+                prettifyOut(GOODBYE);
+                return false;
+        }
```
The `case` blocks should not be indented as per the coding standard. Possibly can consider to add a `default` clause for completeness sake in case in the future there are more cases and for some reason those cases are not yet implemented in this `switch-case`?

Beebeeoii commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        boolean cont = true;
```
Maybe we can rename `cont` to sound more like a boolean? Something like `toCont` or `toContinue`?

Beebeeoii commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+     * Available commands.
+     */
+    public enum Command {
+        LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, BYE
```
`enum` should be written in PascalCase instead of all caps

Beebeeoii commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static final String GREETING = "Hello! I'm Duke\nWhat can I do for you?";
+    private static final String GOODBYE = "Bye. Hope to see you again soon!";
+    private static final ArrayList<Task> TASKS = new ArrayList<>();
```
nice !

Beebeeoii commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,13 @@
+public class ToDo extends Task{
```
There seems to be a missing ` ` (space) between "Task" and "{"

Beebeeoii commented:
Other than some minor nits wrt style, lgtm !

KS2499 committed:
Add save to file functionality

Add methods to auto-save tasks to file.

jerrrren commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
```
If possible wouldn't it be better to use a more descriptive noun for this variable?

jerrrren commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+
 public class Duke {
+    private static final String GREETING = "Hello! I'm Duke\nWhat can I do for you?";
+    private static final String GOODBYE = "Bye. Hope to see you again soon!";
```
Good job following standards for constants

jerrrren commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+    /**
+     * prints border line.
+     */
+    private static void borderLine() {
```
Wouldn't it be more appropriate to name this method as a verb sucha as printBorderLine ?

jerrrren commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+     *
+     * @param inp input string.
+     */
+    private static void prettifyOut(String inp) {
```
Wouldn't naming the method as a verb such as "prettifyOutput" be more appropriate, personally it took me a while to understand what Out means

jerrrren commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,167 @@
+     *
+     * @throws DukeException if input is invalid.
+     */
+    private static boolean executeInput(String inp) throws DukeException {
```
Since this method is a boolean wouldn't it be better name it in a boolean sounding way?

jerrrren reviewed (commented):
In general good job in conforming to java style in terms of formatting, but do take note of some of your boolean names

KS2499 committed:
Modified event and deadline classes

Change the data type of deadline and event attributes.

DateTime datatype is more flexible and convenient in implementing date and time related functions in the future

KS2499 committed:
Merge pull request #1 from KS2499/branch-Level-7

Level-7

KS2499 committed:
Merge branch 'master' into branch-Level-8

KS2499 committed:
Merge pull request #2 from KS2499/branch-Level-8

Modified event and deadline classes

KS2499 committed:
Merge pull request #3 from KS2499/add-gradle-support

Add Gradle support

KS2499 committed:
Made the code more oop

Refactored the code

Extracted out Parser, Ui, Storage, TaskList and DukeCommands

KS2499 committed:
Organise into packages

Organised into packages

KS2499 committed:
Add some junit tests

Added some basic junit tests for task classes

KS2499 committed:
Add find functionality

Add find command to Duke

KS2499 committed:
Merge pull request #4 from KS2499/branch-Level-9

Add find functionality

KS2499 committed:
Add find tasks to TaskList

KS2499 committed:
Add JavaDocs to more functions

KS2499 committed:
Check for any code standard violations

KS2499 committed:
Merge pull request #10 from KS2499/branch-A-CodingStandard

Check for any code standard violations

KS2499 committed:
Merge pull request #9 from KS2499/branch-Level-9

Add find tasks to TaskList

KS2499 committed:
Merge pull request #8 from KS2499/branch-A-JavaDoc

Add JavaDocs to more functions

KS2499 committed:
Update README.md

KS2499 committed:
Add checkstyle to project

KS2499 committed:
Merge pull request #11 from KS2499/branch-A-CheckStyle

Add checkstyle to project

KS2499 committed:
Fix some checkstyle errors

KS2499 committed:
Add GUI and clean-up code

KS2499 committed:
Merge pull request #12 from KS2499/branch-Level-10

Branch-level-10

KS2499 committed:
Edit JavaDoc for Duke::getResponse

KS2499 committed:
Add assert to Storage.java

KS2499 committed:
Add assert to Parser.java

KS2499 committed:
Add assert to ResponseGenerator.java

KS2499 committed:
Merge pull request #13 from KS2499/branch-A-Assertions

Branch A-Assertions

KS2499 committed:
Refactor variable/method names for enhanced readability

KS2499 committed:
Merge pull request #14 from KS2499/branch-A-CodeQuality

Refactor variable/method names for enhanced readability

KS2499 committed:
Add method to clear TaskList

KS2499 committed:
Add method to archive current Tasks

KS2499 committed:
Add method to print archived Tasks

KS2499 committed:
Add archive functionality to Duke and DukeCommand

KS2499 committed:
Merge pull request #15 from KS2499/branch-C-Archive

Branch-C-Archive

KS2499 committed:
Add Ui.png showcasing screenshot of Duke

KS2499 committed:
Fix archive command and Update README.md

Archive command: make archive command not require description

Archive command required description.

Archive command has no need for description.

Update README.md: add relevant features and usage methods in

README is up to date with current product.

KS2499 committed:
Merge pull request #16 from KS2499/branch-A-UserGuide

Branch A-user guide

KS2499 committed:
Fix archive command

Archive was not copying over contents of duke.txt to archive.txt properly

Fixed with help of ChatGPT
